tests/ui/cli/crate-type-flag.rs

```diff
@@ -0,0 +1,57 @@
+//! Check that `rustc`'s `--crate-type` flag accepts `--crate-type=<valid_type>` as well as the
+//! multi-value version `--crate-type=<valid_type_1>,<valid_type_2>`.
+//!
+//! This test does not try to check if the output artifacts are valid.
+
+// FIXME(#132309): add a proper `supports-crate-type` directive.
```

Comment:
This is a known could-be-improved compiletest directive handling aspect, didn't want to fix that in this PR.